File: MediaStreamingPlatform_API/Application/Service/SignalRService.cs
using Microsoft.AspNetCore.SignalR;

public class SignalRService
{
    private readonly IHubContext<MediaHub> _hubContext;

    public SignalRService(IHubContext<MediaHub> hubContext)
    {
        _hubContext = hubContext;
    }

    public async Task NotifyPlaylistStart(int playlistId)
    {
        await _hubContext.Clients.All.SendAsync("PlaylistStarted", new
        {
            PlaylistId = playlistId,
            Action = "START_PLAYLIST"
        });
    }

    public async Task NotifyPlaylistStop()
    {
        await _hubContext.Clients.All.SendAsync("PlaylistStopped", new
        {
            Action = "STOP_PLAYLIST"
        });
    }
}

File: MediaStreamingPlatform_Output_UI/src/Services/SignalRService.ts
import { HubConnection, HubConnectionBuilder } from '@microsoft/signalr';

export class SignalRService {
  private static connection: HubConnection | null = null;
  private static readonly HUB_URL = "http://localhost:5278/mediaHub";

  static async connect(): Promise<void> {
    if (this.connection?.state === 'Connected') return;

    this.connection = new HubConnectionBuilder()
      .withUrl(this.HUB_URL)
      .withAutomaticReconnect()
      .build();

    await this.connection.start();
    console.log('SignalR Connected');
  }

  static async disconnect(): Promise<void> {
    await this.connection?.stop();
    this.connection = null;
  }

  static onPlaylistStarted(callback: (data: { PlaylistId: number, action: string }) => void): void {
    this.connection?.on('PlaylistStarted', callback);
  }

  static off(eventName: string): void {
    this.connection?.off(eventName);
  }
}